### PR TITLE
Run minification for TorchDynamo benchmark models that fail evaluation

### PR DESCRIPTION
**Pull request by awskila**
### Description
As an alternative to PR #105774, which provides a standalone, end-to-end minification script that covers all types of failures and has more functionality, this PR adds the ability to minify models when they fail the eval loop (accuracy checks). Both this PR and the other one can be merged without issue.

### Purpose
The goal is to leverage the minifier to minify models that fail accuracy checks, allowing failed models to be debugged more easily. The ideal use-case is trying to run a model suite on a backend where operator coverage is not known or is limited. If models can compile but fails the eval loop, having the repro script for each model is valuable for any developer that's trying to fix the issue.

### Functionality
- Create minify flag that minifies models when they fail accuracy check
- Produce minified graph for each model, and save it into repro script
- Move repro script to output directory/base Dynamo directory
- Enable functionality for running an entire model suite (Hugging Face, timm, and TorchBench) by prepending model name to repro script

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

### CODE REVIEW

**Comment by ezyang** [comment]
It's just not clear to me that this PR provides enough marginal utility over just putting the two envvars you need to trigger minifier yourself. Remember people have to know the flag exists!

**Comment by awskila** [comment]
> It's just not clear to me that this PR provides enough marginal utility over just putting the two envvars you need to trigger minifier yourself. Remember people have to know the flag exists!

Very fair comment. A couple of points to make: 

1) I'm not sure how many people set those environment variables when running the eval loop for a given model when they don't expect it to fail. I think the current workflow for a lot of people (speaking from my experience) who know about the minifier tool is to run training/inference for a model, and then if it fails accuracy, re-run training on the same model with those environment variables. Better to run with `--minify`, especially when running a list of models, and avoid running the script again.

2)  I made this PR with the idea of aiding those who maybe are running all Hugging Face/TIMM/Torch Bench models at once. Currently, the repro scripts are saved as `repro.py` in the local directory. So if you run the minifier across a whole suite of models, you end up getting the last model's repro script and lose the rest. This PR would fix that problem.

>  Remember people have to know the flag exists!

Should I add this to the [README](https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/README.md)? That might help with this issue.

**Comment by awskila** [comment]
I understand that the variables are documented. The purpose is to run the minifier only on models that fail evaluation, and add repro scripts for these models. There are a lot of flags on the Dynamo benchmark scripts as is - adding one more to help debug failing models with minified subgraphs would be useful to users. 

I know from my own personal experience, after I had models fail evaluation on our custom [Neuron framework](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/), after I got the results for all Hugging Face models, I ended up re-running them one-by-one after setting the minifier variables and renamed the repro scripts to include the model name. Not sure how common my use-case is, but I know it'll help me and others that are working on custom silicon, where model coverage can vary.

**Review by ezyang — APPROVED** [review]

**Comment by awskila** [comment]
Thanks for the approval and merging the change!
